### ISSUE TITLE
learning rate

### ISSUE THREAD

**Issue by yangyu-cmd** [aaroncao06/MedSegMamba#2]
hello author:
    I appreciate your work, can you share the learning rate?

**Comment by aaroncao06** [comment]
Hi,
We used a learning rate scheduler, and the code for it is below:

optimizer = torch.optim.AdamW(
    model.parameters(),
    lr=1e-3,
    betas=(0.9, 0.999),
    eps=1e-8,
    weight_decay=1e-4,
    amsgrad=True
    )
scheduler = CosineAnnealingWarmRestarts(optimizer, T_0 = 1, T_mult = 2, eta_min = 1e-6, last_epoch = -1)

**Comment by yangyu-cmd** [comment]
thanks very much~
I used your network to train for a long time, the loss finally stayed around 1.7 and the dice stayed at 0.82,
 I lowered the learning rate but it didn't work, I think it might have something to do with the dataset, 
do you have any good solution for the loss not to drop? 
Thank you very much for sharing

**Comment by aaroncao06** [comment]
Do you have any more details about your problem? What are you trying to segment, what is your loss function, how many samples do you have, and how many epochs did you train for?

**Comment by yangyu-cmd** [comment]
Thank you for your reply, the dataset I used is liver from Medical Segmentation Decathlon, the loss function used is DiceCELoss, trained for 25000 steps and still continuing, only dice improved to 8.27, but the loss has been maintained without decreasing to 1.7

**Comment by aaroncao06** [comment]
I trained using a combo of weighted cross entropy and dice loss only for the first epoch and switched to just dice loss afterwards, but I think my task involved segmenting many more regions than yours. Also, did you use the patch-based approach like described in our preprint (what is your input image size?), and what dice score are you aiming to reach?

**Comment by yangyu-cmd** [comment]
I choose the input image size as (96,96,96),I train the dice first to reach 85, I am more concerned about the training loss, why can't it go down to about 0.1, is it too little data? Is the data too small?

**Comment by aaroncao06** [comment]
The loss function you have adds cross entropy and dice loss, so it makes sense that it would be higher. If you try using just dice loss, how low does it get? You can also sometimes have “high” cross entropy loss but still good performance, judging from your dice score. The cross entropy loss might just seem high because you are computing it across a whole image. And what is the original image size of what you want to segment (not the model input size)? Do you follow a similar training and inference pipeline to what we described in our preprint?

**Comment by yangyu-cmd** [comment]
I don't have the training method in your papper, I only trained with diceloss and the loss stays around 1.7, the whole size of the dataset I used is (512,512,512), I wouldn't use cross entropy and dice loss in the first round yet, and then dice loss later on, the same dataset I can get the diceloss down to 0.1 with unetr Nearby, I'm confused.

**Comment by aaroncao06** [comment]
Dice loss cannot be greater than 1, so I think there is a mistake somewhere. 
Also, do you use the same inference pipeline as us then? How do you extract the 96x96x96 patches from the input volume for training and inference? Do you then use the same pipeline with unetr?
And what dice score does unetr get?

**Comment by yangyu-cmd** [comment]
My input data in unetr is (8,1,96,96,16), in your network you are using (1,1,96,96,96), in unetr network dice is basically above 85

**Comment by aaroncao06** [comment]
Did you check why you said the dice loss is greater than 1? And it looks like you are training our model with only a batch size of 1, so you can try to increase that. Our model was also designed to segment many classes, so you should probably modify some of the channel sizes if your task only segments a couple.

And what method do you use to extract the 96x96x16 and 96x96x96 patches from the original 512x512x512 image?

**Comment by yangyu-cmd** [comment]
You are correct, I only use diceloss not dice and celoss, the training loss is indeed less than 1
